tinymce-single-react-ui/src/reducers/action.js

```diff
@@ -0,0 +1,57 @@
+import { fromJS } from 'immutable'
+
+// TODO: use redux-actions
+
+// TODO: replace nulls with Option values below
+const initialState = fromJS({
+  focused: false,  // does the editor have focus
+  collapsed: null, // current selection is collapsed
+  bookmark: null,  // current bookmark for restoring the selection
+  node: null,      // current node of the selection (if collapsed) or common ancestor containing the selection)
+  range: null,     // current selection range
+  editorRef: null  // reference to the editor
+})
```

Comment:
storing the blocks in the store: depends on lots of things really
- what we define as blocks
- whether blocks exist in a hierarchy
- assuming the content state is managed in the browser dom

In this Tiny-single example we just deduce the blocks using the existing block search code that looks up from the current selection up to the element which is a direct child of the body. - This implies blocks are a linear sequence.

In the Tiny-singe example we could also look up from the current selection to the nearest enclosing 'block' type element (what ever HTML5 tags or rules those are defined to be).
- This would give a hierarchy of nested blocks following the dom structure
- if we also stored those blocks in the state it would presumably need some sort of tree structure